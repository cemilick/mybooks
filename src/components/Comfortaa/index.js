import {View, Text, StyleSheet} from 'react-native';
import React from 'react';
import {colors} from '../../res/colors';

export default function Index({
  decoration = 'none',
  children,
  type = 'Regular',
  size = 14,
  style,
}) {
  const styles = StyleSheet.create({
    text: {
      color: colors.primaryLight,
      fontFamily: `Comfortaa-${type}`,
      fontSize: size,
      textDecorationLine: decoration,
    },
  });

  return <Text style={[styles.text, style]}>{children}</Text>;
}
