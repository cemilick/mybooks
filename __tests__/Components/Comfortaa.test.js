import React from 'react';
import Comfortaa from '../../src/components/Comfortaa';
import {render} from '@testing-library/react-native';

describe('Comfortaa Testing', () => {
  it('render with 14px size', () => {
    const {getByTestId} = render(<Comfortaa>audy</Comfortaa>);
    expect(getByTestId('text component').props.style[0].fontSize).toEqual(14);
  });

  it('render with white color', () => {
    const {getByTestId} = render(<Comfortaa>audy</Comfortaa>);
    expect(getByTestId('text component').props.style[0].color).toEqual('#FFF');
  });

  it('render with children value', () => {
    const {getByTestId} = render(<Comfortaa>cemilick</Comfortaa>);
    expect(getByTestId('text component').props.children).toEqual('cemilick');
  });

  it('equals with snapshot', () => {
    const {toJSON} = render(<Comfortaa>cemilick</Comfortaa>);
    expect(toJSON()).toMatchSnapshot();
  });
});
