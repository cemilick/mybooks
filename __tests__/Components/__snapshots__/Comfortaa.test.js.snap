// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comfortaa Testing equals with snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#FFF",
        "fontFamily": "Comfortaa-Regular",
        "fontSize": 14,
        "textDecorationLine": "none",
      },
      undefined,
    ]
  }
  testID="text component"
>
  cemilick
</Text>
`;
